### PR TITLE
fix: retry in wait-block-count

### PR DESCRIPTION
**Pull request by maan2003**
why was timeout added here?

### CODE REVIEW

File: fedimint-cli/src/lib.rs [hunk 1]
```diff
@@ -793,23 +793,19 @@ impl FedimintCli {
...
+                        .with_module(wallet.id)
+                        .fetch_consensus_block_count()
+                        .await?;
+                    if count >= target {
+                        break Ok(CliOutput::WaitBlockCount { reached: count });
                     }
-                })
-                .await
-                .map_err_cli_msg("reached timeout")?
+                    task::sleep(Duration::from_millis(100)).await;
```
Comment (dpc) [inline fedimint-cli/src/lib.rs:807]: We should log something on each failure, just to let user know what's going on and that it didn't just hang.

**Comment by dpc** [comment]
See: https://github.com/fedimint/fedimint/pull/4798#discussion_r1548428237

We currently have no retries for errors (`?` returning early). Should we? Normally we don't do it, but given that this command is often used for scripting, maybe we should here?

File: fedimint-cli/src/lib.rs (hunk 1 above)
Comment (elsirion) [reply fedimint-cli/src/lib.rs:807]: That will actually lead to a log every 60s with current server-side timeouts.

**Review by elsirion — DISMISSED** [review]
Idk why there was a timeout in the first place, feel free to add some logging and re-request review :)

**Review by dpc — DISMISSED** [review]

File: fedimint-cli/src/lib.rs (hunk 1 above)
Comment (dpc) [inline fedimint-cli/src/lib.rs:807]: BTW. I wonder if we should close the client when sleeping to unlock the client for other things to run, potentially.

Comment (maan2003) [reply fedimint-cli/src/lib.rs]: updated

**Review by dpc — APPROVED** [review]
LGTM, but I think the linter doesn't like something. **Edit**: doesn't compile. `FnMut` + `async` fun times.

**Comment by maan2003** [comment]
rust needs async closures, closure + async block sucks

**Comment by maan2003** [comment]
I also gave up after 10 minutes, should have trusted you.
made `open` accept `&self`

**Comment by dpc** [comment]
> I also gave up after 10 minutes, should have trusted you.
> made `open` accept `&self`

I couldn't even make that one work ... and now I don't understand why your version works. But I accept you're just better at Rust. :D

**Review by dpc — DISMISSED** [review]

File: fedimint-cli/src/lib.rs [hunk 2]
```diff
@@ -792,25 +793,29 @@ impl FedimintCli {
...
+                    let wallet = client.get_first_module::<WalletClientModule>();
+                    let count = client
+                        .api()
+                        .with_module(wallet.id)
+                        .fetch_consensus_block_count()
+                        .await?;
+                    if count >= target {
+                        Ok(CliOutput::WaitBlockCount { reached: count })
+                    } else {
+                        Err(format_err!("target not reached"))
```
Comment (dpc) [inline fedimint-cli/src/lib.rs:812]: Thought: should we `info!(...)` something here (e.g. target, current, "Block count not reached").

Comment (maan2003) [reply fedimint-cli/src/lib.rs:813]: done

**Review by elsirion — APPROVED** [review]

**Pull request merged**